src/main/kotlin/ecommerce/repository/ProductRepository.kt

```diff
@@ -0,0 +1,51 @@
+package ecommerce.repository
+
+import ecommerce.model.Product
+import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.jdbc.core.RowMapper
+import org.springframework.stereotype.Repository
+import java.sql.ResultSet
+
+@Repository
+class ProductRepository(private val jdbcTemplate: JdbcTemplate) {
+    private val productRowMapper =
+        RowMapper<Product> { rs: ResultSet, _ ->
+            Product(
+                rs.getLong("id"),
+                rs.getString("name"),
+                rs.getDouble("price"),
+                rs.getString("image_url"),
+            )
+        }
+
+    fun findById(id: Long): Product? {
+        return jdbcTemplate.queryForObject("select id, name, price, image_url from products where id = ?", productRowMapper, id)
+    }
+
+    fun getAll(): List<Product> {
+        return jdbcTemplate.query("select id, name, price, image_url from products", productRowMapper)
+    }
+
+    fun createProduct(product: Product) {
+        jdbcTemplate.update(
+            "insert into products (name, price, image_url) values (?,?,?)",
```

Comment:
It would be great if we keep the convention of keeping SQL keywords all uppercase just like you did to `UPDATE` :
```sql
INSERT INTO products (name, price, image_url) VALUES (?,?,?)
```

It makes reading queries a bit easier, and keeping conventions makes the code look more... professional 😎